Duplicati/Library/Snapshots/UsnJournalService.cs

```diff
@@ -60,18 +63,26 @@ public class UsnJournalService
         /// Initialize list of modified files / folder for each volume
         /// </summary>
         /// <param name="emitFilter"></param>
+        /// <param name="fileAttributeFilter"></param>
+        /// <param name="skipFilesLargerThan"></param>
         /// <param name="prevJournalData"></param>
         /// <returns></returns>
-        private Dictionary<string, VolumeData> Initialize(IFilter emitFilter, IEnumerable<USNJournalDataEntry> prevJournalData)
+        private Dictionary<string, VolumeData> Initialize(IFilter emitFilter, FileAttributes fileAttributeFilter, long skipFilesLargerThan,
+            IEnumerable<USNJournalDataEntry> prevJournalData)
         {
             if (prevJournalData == null)
                 throw new UsnJournalSoftFailureException();
 
             var result = new Dictionary<string, VolumeData>();
 
-            // get filter identifying current source filter / sources configuration
+            // get hash identifying current source filter / sources configuration
             // ReSharper disable once PossibleMultipleEnumeration
-            var configHash = (emitFilter == null ? string.Empty : emitFilter.GetFilterHash()) + Utility.Utility.ByteArrayAsHexString(MD5HashHelper.GetHash(m_sources));
+            var configHash = Library.Utility.Utility.ByteArrayAsHexString(MD5HashHelper.GetHash(
+                (emitFilter == null ? string.Empty : emitFilter.ToString()) +
+                string.Join("; ", m_sources) +
+                fileAttributeFilter.ToString() +
+                skipFilesLargerThan.ToString()

```

Comment:
Instead of concatenating everything into a single string, can we create a collection of strings
```
List<string> sourcesAndFilters = new List<string>(this.m_sources)
{   
    emitFilter is null ? String.Empty : emitFilter.ToString(),
    fileAttributeFilter.ToString(),
    skipFilesLargerThan.ToString()
};
```
and provide that to `MD5HashHelper.GetHash`?

With this, I think the comment disabling the ReSharper warning is also no longer needed.

Reply:
From what I can tell, GetHash only takes a single string, not an array of strings. Do you want me to adjust it to accept an array of strings?

Reply:
Did the previous implementation call [this overload](https://github.com/duplicati/duplicati/blob/v2.0.5.101-2.0.5.101_canary_2020-01-23/Duplicati/Library/Utility/MD5HashHelper.cs#L49), which accepts an `IEnumerable<string>`?

Reply:
Sorry, I just saw the overload! Will submit an update.